modules/economy.js

```diff
@@ -1,295 +1,255 @@
-      const status = []
-      if (parseInt(productData.productPrice) <= parseInt(userAccount.amount)) {
-        module.exports.itemActivate(client, member, guild, productData, (iaData, continueBuying) => {
-          status.ia = iaData
-          if (continueBuying) module.exports.updateUserAccount(client, member, guild, parseInt(userAccount.amount) - parseInt(productData.productPrice), () => { })
-          if (continueBuying) status.code = true
-          status.code = status.code || false
-          callback(status)
-        })
-      } else {
-        callback(status.code || false)
-      }
+  updateMemberBalance: (client, member, guild, newBalance, callback) => {
+    // TODO: Update column name from "amount" to "balance"
+    client.pool.query('UPDATE `guildEconomyUserBank` SET `amount` = ? WHERE `member` = ? AND `guild` = ?', [newBalance, member.id, guild.id], (err) => {
+      if (err) client.logError(err)
+      if (callback) callback()
     })
   },
-  updateUserAccount: (client, member, guild, amount, callback) => {
-    client.pool.query('UPDATE `guildEconomyUserBank` SET `amount` = ? WHERE `member` = ? AND `guild` = ?', [amount, member.id, guild.id], (err) => {
+  updateMemberInventory: (client, member, guild, newInventory, callback) => {
+    client.pool.query('UPDATE `guildEconomyUserBank` SET `inventory` = ? WHERE `member` = ? AND `guild` = ?', [newInventory, member.id, guild.id], (err) => {
       if (err) client.logError(err)
-      callback()
+      if (callback) callback()
     })
   },
-  addItemToUser: (client, member, guild, productId, singlebuy, callback) => {
-    module.exports.fetchUserAccount(client, member, guild, (account) => {
-      if (account) {
-        let productQuantity = 1
+  addItemToMemberInventory: (inventoryFromDB, productToAdd, callback) => {
+    // TODO: Replace the property "singlebuy" with "buyOnlyOne".
 
-        singlebuy = singlebuy || false
-        if (singlebuy) productQuantity = -1
+    let productQuantity = 1
 
-        const inventoryData = JSON.parse(account.inventory)
-        if (inventoryData[productId]) {
-          if (!singlebuy) {
-            inventoryData[productId] = parseInt(inventoryData[productId]) + productQuantity
-            if (callback) callback()
-          } else {
-            if (callback) callback('error')
-          }
+    if (productToAdd.singlebuy) productQuantity = -1
+
+    const parsedInventoryFromDB = JSON.parse(inventoryFromDB)
+
+    if (parsedInventoryFromDB[productToAdd.productId]) {
+      if (!productToAdd.singlebuy) parsedInventoryFromDB[productToAdd.productId] = parseInt(parsedInventoryFromDB[productToAdd.productId]) + productQuantity
+      if (productToAdd.singlebuy) parsedInventoryFromDB[productToAdd.productId] = parseInt(parsedInventoryFromDB[productToAdd.productId])
+    } else {
+      parsedInventoryFromDB[productToAdd.productId] = productQuantity
+    }
+
+    Object.keys(parsedInventoryFromDB).forEach(product => {
```

Comment:
Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/alexandrumuntenas/pingu/pullRequest?prid=8657799)
```suggestion
    Object.keys(parsedInventoryFromDB).forEach((product) => {
```